src/main/java/com/aws/greengrass/easysetup/DeviceProvisioningHelper.java

```diff
@@ -289,8 +327,13 @@ private void downloadFileFromURL(String url, File f) throws IOException {
         try (SdkHttpClient client = getSdkHttpClient()) {
             HttpExecuteResponse executeResponse = client.prepareRequest(executeRequest).call();
 
+            int responseCode = executeResponse.httpResponse().statusCode();
+            if (responseCode != HttpURLConnection.HTTP_OK) {
+                throw new IOException("Received invalid response code: " + responseCode);
+            }
+
             try (ReadableByteChannel readableByteChannel = Channels.newChannel(executeResponse.responseBody().get());
-                 FileOutputStream fileOutputStream = new FileOutputStream(f)) {
+                 FileOutputStream fileOutputStream = new FileOutputStream(f, appendFile)) {
```

Comment:
if the existing file doesn't end with \n this will not work properly, right? or would it be cleaned up to work correctly by the duplicate removing code?

Reply:
Looks like it was always writing from the beginning of the file anyway (oops). Changed it so it'll actually append from the end of the file. If the existing file doesn't end with \n and a new cert is appended, it would break cert parsing, but removeDuplicateCertificates does fix it.